### PR TITLE
Support CTEs in CREATE VIEW statements

### LINKED ISSUE #2595: CREATE VIEW statement with CTE throws PRS error
Example DDL:
```
CREATE OR REPLACE VIEW vw_appt_latest AS (
  WITH most_current as (
      SELECT
            da.*
      FROM dim_appt da
      WHERE da.current_appt_id IS NULL
      )
  SELECT * from most_current
)
```

## Expected Behaviour
`lint` should return no PRS errors

## Observed Behaviour
The DDL executes fine on Snowflake, however sqlfluff lint returns:
```
L:   1 | P:   1 |  PRS | Line 1, Position 1: Found unparsable section: 'CREATE OR
                       | REPLACE VIEW vw_appt_latest AS...'
```

## Steps to Reproduce
sqlfluff lint

## Dialect
Snowflake 

## Version

sqlfluff, version 0.9.4

## Configuration

```
[sqlfluff]
verbose = 0
nocolor = False
dialect = snowflake
templater = jinja
rules = L001,L002,L003,L004,L005,L006,L007,L008,L009,L010,L011,L012,L013,L016,L017,L018,L019,L022,L023,L024,L025,L029,L030,L036,L037,L038,L039,L041,L044,L045,L046,L047,L048
exclude_rules = L029,L031
recurse = 0
output_line_length = 80
runaway_limit = 10
ignore_templated_areas = True
ignore = templating
# Comma separated list of file extensions to lint.

# NB: This config will only apply in the root folder.
sql_file_exts = .sql,.dml,.ddl

[sqlfluff:indentation]
indented_joins = False
template_blocks_indent = True

[sqlfluff:templater]
unwrap_wrapped_queries = True

[sqlfluff:templater:jinja]
apply_dbt_builtins = True

[sqlfluff:templater:jinja:macros]
# Macros provided as builtins for dbt projects
dbt_ref = dbt_source = dbt_config = dbt_var = dbt_is_incremental =
# Some rules can be configured directly from the config common to other rules.
[sqlfluff:rules]
tab_space_size = 4
max_line_length = 140
indent_unit = space
comma_style = trailing
allow_scalar = True
single_table_references = consistent
unquoted_identifiers_policy = all

# Some rules have their own specific config.
[sqlfluff:rules:L003]
lint_templated_tokens = True

[sqlfluff:rules:L010]  # Keywords
capitalisation_policy = upper
```

### PR DESCRIPTION
**Pull request by tunetheweb**
### Brief summary of the change made

Fixes #2595

### Are there any other side effects of this change that we should be aware of?
Corrected in quite a few dialects but only added test cases for two

### Pull Request checklist
- [x] Please confirm you have completed any of the necessary steps below.

- Included test cases to demonstrate any code changes, which may be one or more of the following:
  - `.yml` rule test cases in `test/fixtures/rules/std_rule_cases`.
  - `.sql`/`.yml` parser test cases in `test/fixtures/dialects` (note YML files can be auto generated with `tox -e generate-fixture-yml`).
  - Full autofix test cases in `test/fixtures/linter/autofix`.
  - Other.
- Added appropriate documentation for the change.
- Created GitHub issues for any relevant followup/future enhancements if appropriate.

### CODE REVIEW

File: src/sqlfluff/dialects/dialect_ansi.py [hunk 1]
```diff
@@ -3547,6 +3547,7 @@ class CreateTriggerStatementSegment(BaseSegment):
             "PROCEDURE",
             Ref("FunctionNameIdentifierSegment"),
             Bracketed(Ref("FunctionContentsGrammar", optional=True)),
+            optional=True,
```
Comment (barrywhart) [inline src/sqlfluff/dialects/dialect_ansi.py:3550]: This looks like the change you made in the other PR, to support SQLite triggers, I think. Is this related to CTEs in CREATE VIEW somehow?

**Review by barrywhart — APPROVED** [review]
Looks good. One question about whether a change belongs in a different PR.

File: src/sqlfluff/dialects/dialect_ansi.py (hunk 1 above)
Comment (tunetheweb) [reply src/sqlfluff/dialects/dialect_ansi.py]: Yup reverted that. Check the latest version now.
